Unable to call `SectionSetter::set` again due to too-restrictive lifetimes on method signature

Once a `SectionSetter` is obtained by e.g. calling `Ini::with_section()`, it's possible to set multiple key-value pairs by chaining calls to `SectionSetter::set()`. However, it is not possible to call `set()` multiple times on the same variable of type `SectionSetter`. The cause seems to be too restrictive lifetimes in the signature of `SectionSetter::set()`.

### Example

The following code:
```rust
use ini::Ini;

fn main() {
    let mut ini = Ini::new();
    let mut section_setter = ini.with_section(Some("section"));

    // chaining set() calls work
    section_setter.set("a", "b").set("c", "d");
    // but calling set() separately doesn't work
    section_setter.set("e", "f"); // <--- error[E0499]
}
```
fails to compile with the output
```
error[E0499]: cannot borrow `section_setter` as mutable more than once at a time
  --> src/main.rs:10:5
   |
8  |     section_setter.set("a", "b").set("c", "d");
   |     -------------- first mutable borrow occurs here
9  |     // but calling set() separately doesn't work
10 |     section_setter.set("e", "f");
   |     ^^^^^^^^^^^^^^
   |     |
   |     second mutable borrow occurs here
   |     first borrow later used here

For more information about this error, try `rustc --explain E0499`.
error: could not compile `inisec` (bin "inisec") due to 1 previous error
```
using `rust-ini v0.21.0` and `rustc v1.80.0`.

### Probable cause
The signature of `SectionSetter::set()` is
```rust
    pub fn set<K, V>(&'a mut self, key: K, value: V) -> &'a mut SectionSetter<'a>
    where
        K: Into<String>,
        V: Into<String>,
    {...}
```
where `'a` is the lifetime of the `Ini` instance borrowed by `SectionSetter<'a>`.

This signature requires `set()` to borrow `self: SectionSetter` for `'a`, which is the whole duration of the original borrow of  `Ini`. So once `set()` is called on the variable `section_setter`, that variable stays borrowed for its whole lifetime and it's impossible to call `set()` again on it.

### Possible solution
Changing the signature to:
```rust
    pub fn set<'b, K, V>(&'b mut self, key: K, value: V) -> &'b mut SectionSetter<'a>
    where
        K: Into<String>,
        V: Into<String>,
        'a: 'b,
    {...}
```
makes the example code above compile (and all `rust-ini` tests still pass).

What we do here is introduce a new lifetime `'b`, and make the new borrow of `SecionSetter<'a>` only last for `'b`. We also require that `'a` is larger than `'b` with the bound `'a: 'b`.

This way, once the chain of `set()` calls on the variable `section_setter` ends, `'b` expires and the original variable `section_setter` is not borrowed anymore and we can explicitly call `set()` again.

### Other methods
I don't have a good-enough grasp of the whole API surface of `rust-ini` to judge whether this change should be applied to other methods as well. Which is why I'm filing this as an issue rather than a PR.

Most probably this applies to `add()`, `delete()` and `get()` as well.

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -352,10 +352,11 @@ impl<'a> SectionSetter<'a> {
     }
 
     /// Set (replace) key-value pair in this section (all with the same name)
-    pub fn set<K, V>(&'a mut self, key: K, value: V) -> &'a mut SectionSetter<'a>
+    pub fn set<'b, K, V>(&'b mut self, key: K, value: V) -> &'b mut SectionSetter<'a>
     where
         K: Into<String>,
         V: Into<String>,
+        'a: 'b,
     {
         self.ini
             .entry(self.section_name.clone())
@@ -366,10 +367,11 @@ impl<'a> SectionSetter<'a> {
     }
 
     /// Add (append) key-value pair in this section
-    pub fn add<K, V>(&'a mut self, key: K, value: V) -> &'a mut SectionSetter<'a>
+    pub fn add<'b, K, V>(&'b mut self, key: K, value: V) -> &'b mut SectionSetter<'a>
     where
         K: Into<String>,
         V: Into<String>,
+        'a: 'b,
     {
         self.ini
             .entry(self.section_name.clone())
@@ -380,7 +382,11 @@ impl<'a> SectionSetter<'a> {
     }
 
     /// Delete the first entry in this section with `key`
-    pub fn delete<K: AsRef<str>>(&'a mut self, key: &K) -> &'a mut SectionSetter<'a> {
+    pub fn delete<'b, K>(&'b mut self, key: &K) -> &'b mut SectionSetter<'a>
+    where
+        K: AsRef<str>,
+        'a: 'b,
+    {
         for prop in self.ini.section_all_mut(self.section_name.as_ref()) {
             prop.remove(key);
         }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -2348,9 +2348,7 @@ bar = f
     fn add_properties_api() {
         // Test duplicate properties in a section
         let mut ini = Ini::new();
-        ini.with_section(Some("foo"))
-            .add("a", "1")
-            .add("a", "2");
+        ini.with_section(Some("foo")).add("a", "1").add("a", "2");
 
         let sec = ini.section(Some("foo")).unwrap();
         assert_eq!(sec.get("a"), Some("1"));
@@ -2358,9 +2356,7 @@ bar = f
 
         // Test add with unique keys
         let mut ini = Ini::new();
-        ini.with_section(Some("foo"))
-            .add("a", "1")
-            .add("b", "2");
+        ini.with_section(Some("foo")).add("a", "1").add("b", "2");
 
         let sec = ini.section(Some("foo")).unwrap();
         assert_eq!(sec.get("a"), Some("1"));
@@ -2368,9 +2364,7 @@ bar = f
 
         // Test string representation
         let mut ini = Ini::new();
-        ini.with_section(Some("foo"))
-            .add("a", "1")
-            .add("a", "2");
+        ini.with_section(Some("foo")).add("a", "1").add("a", "2");
         let mut buf = Vec::new();
         ini.write_to(&mut buf).unwrap();
         let ini_str = String::from_utf8(buf).unwrap();

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -2636,4 +2636,37 @@ x3 = nb
             ]
         );
     }
+
+    #[test]
+    fn section_setter_chain() {
+        // fix issue #134
+
+        let mut ini = Ini::new();
+        let mut section_setter = ini.with_section(Some("section"));
+
+        // chained set() calls work
+        section_setter.set("a", "1").set("b", "2");
+        // separate set() calls work
+        section_setter.set("c", "3");
+
+        assert_eq!("1", section_setter.get("a").unwrap());
+        assert_eq!("2", section_setter.get("b").unwrap());
+        assert_eq!("3", section_setter.get("c").unwrap());
+
+        // overwrite values
+        section_setter.set("a", "4").set("b", "5");
+        section_setter.set("c", "6");
+
+        assert_eq!("4", section_setter.get("a").unwrap());
+        assert_eq!("5", section_setter.get("b").unwrap());
+        assert_eq!("6", section_setter.get("c").unwrap());
+
+        // delete entries
+        section_setter.delete(&"a").delete(&"b");
+        section_setter.delete(&"c");
+
+        assert!(section_setter.get("a").is_none());
+        assert!(section_setter.get("b").is_none());
+        assert!(section_setter.get("c").is_none());
+    }
 }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -389,7 +389,7 @@ impl<'a> SectionSetter<'a> {
     }
 
     /// Get the entry in this section with `key`
-    pub fn get<K: AsRef<str>>(&'a mut self, key: K) -> Option<&'a str> {
+    pub fn get<K: AsRef<str>>(&'a self, key: K) -> Option<&'a str> {
         self.ini
             .section(self.section_name.as_ref())
             .and_then(|prop| prop.get(key))

